File: backend/lambdas/broadcast-lambda/src/main/java/com/artcorner/erp/dto/response/orders/OrderResponseForCustomer.java
package com.artcorner.erp.dto.response.orders;

import com.artcorner.erp.entities.orders.OrderStatus;
import lombok.Builder;
import lombok.Data;

import java.math.BigDecimal;
import java.time.LocalDateTime;
import java.util.List;
import java.util.UUID;

@Data
@Builder
public class OrderResponseForCustomer {
    private Long orderId;
    private UUID customerId;
    private OrderStatus orderStatus;
    private BigDecimal totalAmount;
    private LocalDateTime createdAt;
    private List<OrderItemResponse> orderItemResponses;

    @Data
    @Builder
    public static class OrderItemResponse {
        private Long itemId;
        private Long productId;
        private String productName;
        private int quantity;
        private BigDecimal price;
    }

}

File: frontend/Customers/src/features/orders/types/OrderResponseForCustomer.ts
export interface OrderResponseForCustomer {
  orderId: number;
  customerId: string;
  orderStatus: OrderStatus;
  totalAmount: number;
  createdAt: Date | string;
  orderItemResponses: OrderItemResponse[];
}

export type OrderStatus =
  | "PENDING"
  | "CANCELED"
  | "ACCEPTED"
  | "PROCESSING"
  | "READY"
  | "DELIVERING"
  | "COMPLETED";

export interface OrderItemResponse {
  itemId: number;
  productId: number;
  productName: string;
  quantity: number;
  price: number;
}
